parser/metadata_test.go

```diff
@@ -376,51 +376,109 @@ func TestGetAndInsertGeolocationIPStruct(t *testing.T) {
 
 }
 
-func TestAddMetaDataNDTConnSpec(t *testing.T) {
+func testTime() time.Time {
 	tst, _ := time.Parse(time.RFC3339, "2002-10-02T15:00:00Z")
-	tests := []struct {
-		spec      schema.Web100ValueMap
-		timestamp time.Time
-		url       string
-		res       schema.Web100ValueMap
-	}{
-		{
-			spec: func() schema.Web100ValueMap {
-				spec := schema.EmptyConnectionSpec()
-				spec["client_ip"] = "127.0.0.1"
-				spec["server_ip"] = "1.0.0.127"
-				return spec
-			}(),
-			timestamp: tst,
-			url:       "/10583?",
-			res: func() schema.Web100ValueMap {
-				spec := schema.EmptyConnectionSpec()
-				spec["client_ip"] = "127.0.0.1"
-				spec["server_ip"] = "1.0.0.127"
-				geoc := spec.Get("client_geolocation")
-				geoc["country_code"] = "US"
-				geoc["country_code3"] = "USA"
-				geoc["country_name"] = "United States of America"
-				geoc["region"] = "NY"
-				geoc["city"] = "Scarsdale"
-				geoc["area_code"] = int64(10583)
-				geoc["postal_code"] = "10583"
-				geoc["latitude"] = float64(41.0051)
-				geoc["longitude"] = float64(73.7846)
-				geos := spec.Get("server_geolocation")
-				geos["country_code"] = "US"
-				geos["country_code3"] = "USA"
-				geos["country_name"] = "United States of America"
-				geos["region"] = "NY"
-				geos["city"] = "Scarsdale"
-				geos["area_code"] = int64(10584)
-				geos["postal_code"] = "10584"
-				geos["latitude"] = float64(41.0051)
-				geos["longitude"] = float64(73.7846)
-				return spec
-			}(),
-		},
+	return tst
+}
+
+var tests = []struct {
```

Comment:
Is this an idiomatic Go thing that I'll have to get used to, or should I regard it as worryingly janky?

Reply:
It is idiomatic go testing structure.
Unless you mean the anonymous struct, in which case, it is general idiomatic go.